ios/REANodesManager.m

```diff
@@ -553,4 +595,36 @@ - (NSString *)obtainProp:(nonnull NSNumber *)viewTag propName:(nonnull NSString
   return result;
 }
 
+- (void)tryToFlushUpdateBuffer
+{
+  [_componentUpdateBufferLock lock];
```

Comment:
What would be helpful is if we add assertions to methods that'd indicate on which thread given method is allowed to be executed. (e.g., `RCTAssertMainQueue`)